Automattic/wp-calypso
# Plans 2023: Split and export plan-features and plan-comparison grids, et al

Related to https://github.com/Automattic/wp-calypso/issues/78266

## Proposed Changes

* Converts the main package component `plans-features-2023-grid` into an index of individual parts, exported and assembled at consuming end
* `ComparisonGrid` and `FeaturesGrid` are moved into respective folders and exported. No title and plan-type-selector embedded
* A "shared" folder is introduced to start moving shared components into

## Testing Instructions

- Confirm `/start` and `/plans/[ paid | free ]` render as before
- Confirm tailored flows (newsletter, wooexpress) work as before
- Confirm plan-term toggle works correctly
- Confirm Odie AI assistant works like before.
    - Visit `/plans` with `?flags=odie`
    - Scroll to the bottom until `Compare Plans` shows. Odie widget should slide in from the side (and stay present until reload - as per original)
- Confirm no performance degradation, or minimal (some might be perceived due to more selectors moving to `plans-features-main`). Confirm over live-image vs staging, or trunk vs this-branch. If not critical, we handle improvements in a follow-up



## Pre-merge Checklist
